common/notification/channels/push.ts

```diff
@@ -0,0 +1,52 @@
+import { Push_subscription as PushSubscription } from '../../../graphql/generated';
+import { getLogger } from '../../logger/logger';
+import { prisma } from '../../prisma';
+import { User } from '../../user';
+import { RedundantError } from '../../util/error';
+
+const logger = getLogger('Push');
+
+// ------------ Subscription Management -------------------------
+// Users can register and remove push subscriptions, to which push notifications are sent
+
+export { PushSubscription };
+export type CreatePushSubscription = Omit<PushSubscription, 'id' | 'userID'>;
+
+export async function getPushSubscriptions(user: User) {
+    return await prisma.push_subscription.findMany({
+        where: {
+            userID: user.userID,
+            // Exclude expired subscriptions:
+            OR: [{ expirationTime: { gte: new Date() } }, { expirationTime: null }],
```

Comment:
We should later add a job to remove these, right?

Reply:
right, but I don't think it is that important. Also not sure if browsers actually set this - It's defined by the standard so I implemented it ...